docs(grid):update autofit grid kb (#2076)
telerik/blazor-docs: knowledge-base/grid-autofit-columns.md

While the autofitting feature includes fitting the headers, it also targets the data cells and the footers, so it needs all the rows in the process.

> There are two possible implementations to work this around and autofit the Grid columns as the component initializes. They depend on whether the Grid is data-bound with the [`Data` parameter](#data-parameter) or the [`OnRead` event](#onread-event). In both cases, the Grid [`Resizable` parameter]({%slug components/grid/columns/resize%}) has to be `true`.

Reviewer: ```suggestion
There are two possible implementations to work this around and autofit the Grid columns as the component initializes. They depend on whether the Grid is data-bound with the [`Data` parameter](#data-parameter) or the [`OnRead` event](#onread-event). In both cases, the Grid [`Resizable` parameter]({%slug components/grid/columns/resize%}) must be set to `true`.
```

### Data Parameter

To AutoFit the Grid columns on initial load of the component you have to:
* Use the Grid [`Class` parameter]({%slug grid-overview%}#grid-parameters) to identify the Grid that you want to autofit.
* When the Grid is rendered for the first time, call a JS function in the `OnAfterRenderAsync` method.

> * The JS function `observeTarget` in the second tab in the example below sets up the `MutationObserver`. This is a JavaScript tool, which listens for DOM changes.

Reviewer 2: This bullet item doesn't relate to the others - it's not something that the developer needs to do. It provides additional information for the last bullet point on the list. 

Consider reworking the last two bullet points into something like this:

```
* Listen for DOM changes and trigger the `AutoFitAllColumns` method when the nodes in the content of the Grid have mutated.

The **JavaScript** tab in the following example shows how to use an `observeTarget` JS function to set up the `MutationObserver` tool to listen for DOM changes.
```

> * Use the `MutationObserver` to trigger the `AutoFitAllColumns` method when the nodes in the content of Grid have mutated (rendered in this case).

Reviewer: ```suggestion
* Use the `MutationObserver` to trigger the `AutoFitAllColumns` method when the table rows in the Grid have rendered.
```

While the autofitting feature includes fitting the headers, it also targets the data cells and the footers, so it needs all the rows in the process.

There are two possible implementations to work this around and autofit the Grid columns as the component initializes. They depend on whether the Grid is data-bound with the [`Data` parameter](#data-parameter) or the [`OnRead` event](#onread-event). In both cases, the Grid [`Resizable` parameter]({%slug components/grid/columns/resize%}) must be set to `true`.

### Data Parameter

> To autofit the Grid columns on the initial load of the component:
> * Use the Grid [`Class` parameter]({%slug grid-overview%}#grid-parameters) to identify the Grid that you want to autofit.
> * When the Grid is rendered for the first time, call a JS function in the `OnAfterRenderAsync` to observe the Grid.
> * Listen for DOM changes using JS and invoke the `AutoFitAllColumns` method when the table rows in the Grid have rendered.

Reviewer: ```suggestion
To autofit the Grid columns on the initial load of the component:

* Use the Grid [`Class` parameter]({%slug grid-overview%}#grid-parameters) to identify the Grid that you want to autofit.
* When the Grid is rendered for the first time, call a JavaScript function in `OnAfterRenderAsync` to observe the Grid.
* Listen for DOM changes using JavaScript and invoke the `AutoFitAllColumns` method after the table rows in the Grid have rendered.
```
